libraries/xcode.rb

```diff
@@ -9,9 +9,15 @@ def installed?(semantic_version)
         installed_xcodes.include?(semantic_version)
       end
 
-      def bundle_version_correct?
-        xcode_version = '/Applications/Xcode.app/Contents/version.plist CFBundleShortVersionString'
-        node['macos']['xcode']['version'] == shell_out("defaults read #{xcode_version}").stdout.strip
+      def find_apple_id(data_bag_retrieval, node_credential_attributes)
+        if node_credential_attributes
+          { 'apple_id' => node_credential_attributes['user'],
+            'password' => node_credential_attributes['password'] }
+        else
+          data_bag_retrieval.call
+        end
+      rescue Net::HTTPServerException
```

Comment:
Is there a chance that we could get this exception under any other circumstances? If so, maybe our message should say something more generic, but include a hint that it is probably missing credentials that caused the error. IIRC, we've gotten errors before due to Apple's servers not responding, but I'm not sure if that's this exact exception.

Reply:
This `rescue` will only save exceptions arising from `data_bag_retrieval.call`, which is before we ever reach out to Apple.